### PR TITLE
Filter OH requests by status

### PR DESCRIPTION
**Pull request by eddierubeiz**
Ref #2478.
See also the "add pagination" PR linked below

### CODE REVIEW

File: app/controllers/admin/oral_history_access_requests_controller.rb [hunk 1]
```diff
@@ -1,12 +1,28 @@
 # STAFF FACING
 # Public facing actions are in app/controllers/oral_history_access_requests_controller.rb
 require 'csv'
+require "active_support/all"
```
Comment (jrochkind) [inline app/controllers/admin/oral_history_access_requests_controller.rb:4]: You should not need to require `active_support/all` here, as we're in a Rails app, it should already be there I think?

Curious why you added this; if it's actually needed, let's documetn why? (And it makes me worry what's going wrong or what I don't understand if it's actually needed!)

File: app/controllers/admin/oral_history_access_requests_controller.rb [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+
+  def status_filter_options
+    statuses = Admin::OralHistoryAccessRequest.delivery_statuses.keys
+    capital_statuses = statuses.map {|s|  ActiveSupport::Inflector.titleize(s) }
+    items =  ([['Any', 'any']] + capital_statuses.zip(statuses)).to_h
+    helpers.options_for_select(
+      items,
+      params.dig(:query, :status) || ""
+    )
   end
```
Comment (jrochkind) [inline app/controllers/admin/oral_history_access_requests_controller.rb:24]: should we bother memo'izing this so it isn't re-calculated every time it's called in display? not sure if it matters.

Comment (eddierubeiz) [reply app/controllers/admin/oral_history_access_requests_controller.rb:24]: I think that's a good idea.

**Comment by eddierubeiz** [comment]
Also adding some tests for the page itself (heretofore we only had tests for the csv download)

File: app/views/admin/oral_history_access_requests/index.html.erb [hunk 3]
```diff
@@ -55,7 +72,7 @@
           <td class="col-sm-4">
              <%= oh_request.intended_use %>
           </td>
-          <td class="col-sm-1">
+          <td class="col-sm-1 request-list-delivery-status">
```
Comment (jrochkind) [inline app/views/admin/oral_history_access_requests/index.html.erb:75]: You added a class name even though you didn't add any styles targeting it? Doesn't hurt though!

**Review by jrochkind — APPROVED** [review]

File: app/views/admin/oral_history_access_requests/index.html.erb (hunk 3 above)
Comment (eddierubeiz) [reply app/views/admin/oral_history_access_requests/index.html.erb:75]: Yeah. I use it in the tests and it looks harmless enough.

**Pull request merged**